engine/uieventmanager.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "uielement.h"
+
+#include <memory>
+#include <vector>
+
+namespace pong
+{
+
+class UIEventManager
+{
+public:
+    void ProcessEvents(const std::vector<std::unique_ptr<UIElement>>& uiElements);
```

Comment:
Not sure if you use this type a lot, but could consider aliasing it

```
using UiElementCollection = std::vector<.....>
```

May save you some eye damage